mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -67,14 +67,75 @@ macro_rules! event {
     }
 }
 
-#[derive(Debug, Clone, Default)]
+#[derive(Debug, Clone)]
 pub struct S3ClientConfig {
-    pub auth_config: S3ClientAuthConfig,
-    pub throughput_target_gbps: Option<f64>,
-    pub part_size: Option<usize>,
-    pub endpoint: Option<Endpoint>,
-    pub user_agent_prefix: Option<String>,
-    pub request_payer: Option<String>,
+    auth_config: S3ClientAuthConfig,
+    throughput_target_gbps: f64,
+    part_size: usize,
+    endpoint: Option<Endpoint>,
+    user_agent_prefix: Option<String>,
+    request_payer: Option<String>,
+}
+
+impl Default for S3ClientConfig {
+    fn default() -> Self {
+        Self {
+            auth_config: Default::default(),
+            throughput_target_gbps: 10.0,
+            part_size: 8 * 1024 * 1024,
```

Comment:
Default values are duplicated here and in mountpoint main. Not sure what to do about it, except maybe a comment here or there.